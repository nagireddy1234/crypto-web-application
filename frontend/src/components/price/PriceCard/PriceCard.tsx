type PriceCardProps = {
  pair: string;
  price: number;
};

export const PriceCard = ({ pair, price }: PriceCardProps) => {
  return (
    <div
      key={pair}
      className="bg-gradient-to-br from-blue-600 to-green-500 p-6 rounded-lg shadow-xl transition-transform duration-300 ease-out transform hover:scale-105 hover:shadow-2xl"
    >
      <h2 className="text-lg font-semibold text-white tracking-wide mb-2">
        {pair}
      </h2>
      <p className="text-4xl font-extrabold text-white tracking-tight">
        {price % 1 === 0 ? price : price.toFixed(6)}
      </p>
    </div>
  );
};
