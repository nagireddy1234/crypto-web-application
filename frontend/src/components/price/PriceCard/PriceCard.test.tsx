// src/components/PriceCard.test.tsx
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { PriceCard } from "./PriceCard";

describe("PriceCard", () => {
  it("renders the pair and price correctly with decimals", () => {
    render(<PriceCard pair="USD" price={100} />);

    expect(screen.getByText("USD")).toBeInTheDocument();
    expect(screen.getByText("100")).toBeInTheDocument();
  });

  it("renders the pair and price correctly without decimals", () => {
    render(<PriceCard pair="USD" price={10.7834} />);

    expect(screen.getByText("USD")).toBeInTheDocument();
    expect(screen.getByText("10.783400")).toBeInTheDocument();
  });

  it("applies the correct font styles", () => {
    render(<PriceCard pair="EUR" price={200} />);

    const pairElement = screen.getByText("EUR");
    const priceElement = screen.getByText("200");

    expect(pairElement).toHaveClass("font-semibold");
    expect(priceElement).toHaveClass("font-extrabold");
  });

  it("should match snapshots", () => {
    const { container } = render(<PriceCard pair="GBP" price={300} />);
    expect(container).toMatchSnapshot();
  });
});
