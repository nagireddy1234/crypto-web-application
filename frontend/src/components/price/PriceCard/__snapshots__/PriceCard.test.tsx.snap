// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PriceCard > should match snapshots 1`] = `
<div>
  <div
    class="bg-gradient-to-br from-blue-600 to-green-500 p-6 rounded-lg shadow-xl transition-transform duration-300 ease-out transform hover:scale-105 hover:shadow-2xl"
  >
    <h2
      class="text-lg font-semibold text-white tracking-wide mb-2"
    >
      GBP
    </h2>
    <p
      class="text-4xl font-extrabold text-white tracking-tight"
    >
      300
    </p>
  </div>
</div>
`;
